runtime/auth/signing/aws-signing-common/common/src/aws/smithy/kotlin/runtime/auth/signing/awssigning/common/middleware/AwsSignerContextMiddleware.kt

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+package aws.smithy.kotlin.runtime.auth.signing.awssigning.common.middleware
+
+import aws.smithy.kotlin.runtime.auth.signing.awssigning.common.AwsSigner
+import aws.smithy.kotlin.runtime.auth.signing.awssigning.common.AwsSigningAttributes
+import aws.smithy.kotlin.runtime.http.operation.InitializeMiddleware
+import aws.smithy.kotlin.runtime.http.operation.OperationRequest
+import aws.smithy.kotlin.runtime.io.Handler
+
+class AwsSignerContextMiddleware<Request, Response>(val signer: AwsSigner) : InitializeMiddleware<Request, Response> {
```

Comment:
This is here to stash the `AwsSigner` so it can be retrieved later in situations where the client config is not available, namely in [`Flow<Message>.sign()`](https://github.com/awslabs/aws-sdk-kotlin/blob/1a21a4773eb9380701e49997a82ae24786138a40/aws-runtime/protocols/aws-event-stream/common/src/aws/sdk/kotlin/runtime/protocol/eventstream/EventStreamSigning.kt#L33) which is called from a serializer. I suppose good Kdocs on the class would've indicated that if I'd wrote them. 😅

I briefly tried threading client config into the serializer before realizing it was turning into A Big Deal. I don't love this either so I'm open to suggestions.

Reply:
I see, well event streams still need some massaging anyway and if that's the only use case right now we can just put a FIXME/TODO in the `Flow<>.sign()` implementation, punt on figuring out config, and remove this for now. The most probably place to do it would be in setting up the operation where we populate context but would have to sit down and think on it.